addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,171 @@
+from markupsafe import Markup, escape
+from odoo import models, _
+
+
+class SurveyUser_Input(models.Model):
+    _inherit = "survey.user_input"
+
+    def _mark_done(self):
+        """ Check if we need to create a lead on answer submission, except for live sessions.
+
+            (note: Here, we are going to filter the surveys submitted at the end by the user according to type.
+            However, the live_session and custom types are present in the filter below.
+            Their presence here corresponds to the case where a survey has simply been shared
+            (as for the classic 'survey' type) without there being a live session.
+            In this case, it is when the user has finally submitted the survey.
+            Otherwise, if the submission is due from the host of a live session, this function is not called.
+            So, please refer to the action_end_session function in survey.survey.)
+        """
+        super()._mark_done()
+
+        # Generate lead
+        user_inputs = self.filtered(lambda user_input:
+            user_input.survey_id.survey_type in ['survey', 'live_session', 'custom']
+        )
+        user_inputs._create_leads_if_generative_answers()
+
+    def _create_leads_if_generative_answers(self):
+        """ This method filters the inputs to create leads and prepares the field
+            data for each input accordingly. """
```

Comment:
`filter` is a bit ambiguous (could be filter out, if you see what I mean)

Suggestion:

"This method filters the user inputs in self to only keep those that will create a lead. After a step of data preparation, create related leads in batch."